src/app/hooks/useLocation/index.test.tsx

```diff
@@ -1,8 +1,19 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
 import { renderHook } from '@testing-library/react-hooks';
 import { fireEvent } from '@testing-library/react';
+import { useState } from 'react';
 import useLocation, * as useLocationObj from '.';
 
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useState: jest.fn(),
+}));
+
 describe('useLocation', () => {
+  beforeEach(() => {
+    // @ts-ignore
+    useState.mockImplementation(jest.requireActual('react').useState);
```

Comment:
You can cast `useState` to `jest.Mock` like: `(useState as jest.Mock)` and remove the `// @ts-ignore` comments

Reply:
Good shout!